### DISCUSSION TITLE
Messages correlation

### DISCUSSION THREAD

**Discussion by de-Bat** [datalust/seq-tickets#1704] [Help] [answered]
Hi
Is it possible to correlate 2 messages ( based on an id property ). And do some Calcukationd, for example: duration between the two messages, how many pairs per...
Thanks,
Ron

**Comment by liammclennan** [comment, marked as answer]
Hi @de-Bat,

If you know the id property you can filter to those events. If you don't know the id property you can group by that id property. In either case you can then perform aggregations on the set of events.

https://docs.datalust.co/docs/aggregate-operators

If you have two events selected then the duration between the two will be `max(@Timestamp) - min(@Timestamp)`.

**Reply by de-Bat** [reply]
Hi, 
Thanks alot for the quick answer.
Sorry, but I dont have much experience with sql. I did a group by on the correlating property but i do not know how to find the time diiference between the events...
My scenario is sending a message  between 2 processes. 
One is publishinnd the other recieves. We use a unique id to identify the message. I would like to know the time between publishing and receiving for each message.
Thanks again, 
Ron

**Reply by liammclennan** [reply]
Something like:

```
select TotalMilliseconds(max(@Timestamp) - min(@Timestamp)) as Elapsed, count(*) as count
from stream
group by CorrelatingProperty
having count = 2
```

The `having` clause is there because this only makes sense when there are two events with the same correlating property value.

**Reply by de-Bat** [reply]
Thanks, It works!
Ron
